File: SNS.Web/src/app/content-management/posts/contracts/post-details.dto.ts
import { CommunitySnapshotDto } from '../../../shared/contracts/community-snapshot.dto';
import { ProfileSnapshotDto } from '../../../profiles/profiles/contracts/profile-snapshot.dto';
import { Paged } from '../../../shared/contracts/paged';
import { CommentSummaryDto } from '../../comments/contracts/comment-summary.dto';
import { PostMediaDto } from './post-media.dto';
import { PostMentionDto } from './post-mention.dto';

export enum PostType {
    Profile = 'Profile',
    Community = 'Community'
}

export enum PostStatus {
    Published = 'Published',
    Archived = 'Archived',
    UnderReview = 'UnderReview',
    Pending = 'Pending',
    Rejected = 'Rejected'
}

export interface PostDetailsDto {
    id: string;
    title: string;
    content: string;
    isPinned: boolean;
    type: PostType;
    status?: PostStatus | null;
    engagementScore: number;
    saveCount: number;
    createdAt: string;
    updatedAt: string;
    author: ProfileSnapshotDto;
    community?: CommunitySnapshotDto | null;
    media: PostMediaDto[];
    comments: Paged<CommentSummaryDto>;
    tags: string[];
    reactionCount: number;
    mentions: PostMentionDto[];
}

File: backend/SNS.Application/ContentManagement/Posts/Posts/Contracts/PostDetailsDto.cs
using SNS.Application.ContentManagement.Comments.Comments.Contracts;
using SNS.Application.ContentManagement.Communities.Communities.Contracts;
using SNS.Application.ContentManagement.Posts.PostMentions.Contracts;
using SNS.Application.Profiles.Profiles.Contracts;
using SNS.Application.Shared.DTOs;
using SNS.Domain.ContentManagement.Posts.Enums;

namespace SNS.Application.ContentManagement.Posts.Posts.Contracts;

public sealed record PostDetailsDto(
    Guid Id,
    string Title,
    string Content,
    bool IsPinned,
    PostType Type,
    PostStatus? Status,
    int EngagementScore,
    int SaveCount,
    DateTime CreatedAt,
    DateTime UpdatedAt,
    ProfileSnapshotDto Author,
    CommunitySnapshotDto? Community,
    List<PostMediaDto> Media,
    Paged<CommentSummaryDto> Comments,
    List<string> Tags,
    int ReactionCount,
    List<PostMentionDto> Mentions
);
